tests/test_module_loader.py

```diff
@@ -616,10 +615,11 @@ def test_project_requirements_dont_overwrite_core_requirements_source(
     jinja2_version_before = active_env.get_installed_packages()["Jinja2"].base_version
 
     # Install the module
-    with pytest.raises(InvalidModuleException) as e:
-        ModuleTool().install(editable=False, path=module_path)
+    with pytest.raises(ConflictingRequirements) as e:
+        ModuleTool().build(path=module_path)
+        env.process_env.install_from_source([env.LocalPackagePath(path=module_path, editable=False)])
```

Comment:
I think only the `install_from_source` is expected to raise a `ConflictingRequirements` ? If so, I would move the `build` part outside of the `with` block to make it clearer